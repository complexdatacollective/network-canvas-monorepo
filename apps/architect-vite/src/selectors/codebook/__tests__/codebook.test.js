/* eslint-env jest */

import testState from '../../../__tests__/testState.json';
import {
  getVariableOptionsForSubject,
  getAllVariablesByUUID,
  makeGetVariable,
} from '..';

describe('codebook selectors', () => {
  describe('getVariableOptionsForSubject()', () => {
    it('extracts variables for nodeType into options list for node entity', () => {
      const subject = {
        type: 'bar',
        entity: 'node',
      };

      const result = getVariableOptionsForSubject(
        testState,
        subject,
      );

      expect(result).toMatchSnapshot();
    });

    it('extracts variables for nodeType into options list for ego entity', () => {
      const subject = {
        type: undefined,
        entity: 'ego',
      };

      const result = getVariableOptionsForSubject(
        testState,
        subject,
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('getAllVariablesByUUID()', () => {
    it('returns all variables by UUID', () => {
      const result = getAllVariablesByUUID(testState);

      expect(result).toMatchSnapshot();
    });

    it('handles missing codebook', () => {
      const result = getAllVariablesByUUID({});

      expect(result).toMatchSnapshot();
    });

    it('handles missing nodeTypes', () => {
      const result = getAllVariablesByUUID({
        codebook: {
          edgeTypes: {},
          ego: {},
        },
      });

      expect(result).toMatchSnapshot();
    });

    it('handles missing edgeTypes', () => {
      const result = getAllVariablesByUUID({
        codebook: {
          nodeTypes: {},
          ego: {},
        },
      });

      expect(result).toMatchSnapshot();
    });

    it('handles missing ego', () => {
      const result = getAllVariablesByUUID({
        codebook: {
          nodeTypes: {},
          edgeTypes: {},
        },
      });

      expect(result).toMatchSnapshot();
    });

    it('handles missing variables', () => {
      const result = getAllVariablesByUUID({
        codebook: {
          nodeTypes: {
            foo: {},
          },
          edgeTypes: {
            bar: {},
          },
          ego: {},
        },
      });

      expect(result).toMatchSnapshot();
    });
  });

  describe('makeGetVariable()', () => {
    it('returns a variable by UUID', () => {
      const result = makeGetVariable('foo')(testState);

      expect(result).toMatchSnapshot();
    });

    it('returns null if variable is not found', () => {
      const result = makeGetVariable('not found')(testState);

      expect(result).toBeNull();
    });

    it('returns error if codebook is not found', () => {
      expect(() => makeGetVariable('foo')({})).toThrow();
    });
  });
});
