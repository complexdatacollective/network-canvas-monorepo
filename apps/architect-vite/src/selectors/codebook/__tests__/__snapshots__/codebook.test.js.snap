// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codebook selectors getAllVariablesByUUID() handles missing codebook 1`] = `Object {}`;

exports[`codebook selectors getAllVariablesByUUID() handles missing edgeTypes 1`] = `Object {}`;

exports[`codebook selectors getAllVariablesByUUID() handles missing ego 1`] = `Object {}`;

exports[`codebook selectors getAllVariablesByUUID() handles missing nodeTypes 1`] = `Object {}`;

exports[`codebook selectors getAllVariablesByUUID() handles missing variables 1`] = `Object {}`;

exports[`codebook selectors getAllVariablesByUUID() returns all variables by UUID 1`] = `Object {}`;

exports[`codebook selectors getVariableOptionsForSubject() extracts variables for nodeType into options list for ego entity 1`] = `
Array [
  Object {
    "isUsed": false,
    "label": "SIERRA",
    "type": "text",
    "value": "sierra",
  },
  Object {
    "isUsed": false,
    "label": "TANGO",
    "type": "text",
    "value": "tango",
  },
]
`;

exports[`codebook selectors getVariableOptionsForSubject() extracts variables for nodeType into options list for node entity 1`] = `
Array [
  Object {
    "isUsed": true,
    "label": "ALPHA",
    "type": "text",
    "value": "alpha",
  },
  Object {
    "isUsed": true,
    "label": "BRAVO",
    "type": "text",
    "value": "bravo",
  },
  Object {
    "isUsed": false,
    "label": "CHARLIE",
    "type": "location",
    "value": "charlie",
  },
  Object {
    "isUsed": true,
    "label": "1234-1234-4",
    "type": "text",
    "value": "1234-1234-4",
  },
  Object {
    "isUsed": true,
    "label": "1234-1234-5",
    "type": "text",
    "value": "1234-1234-5",
  },
]
`;

exports[`codebook selectors makeGetVariable() returns a variable by UUID 1`] = `null`;
